File: src/main/java/com/example/storage/service/impl/EmployeeServiceImpl.java
package com.example.storage.service.impl;

import com.example.storage.controller.dto.EmployeeDto;
import com.example.storage.exception.EmployeeNotFoundException;
import com.example.storage.repository.EmployeeRepository;
import com.example.storage.service.EmployeeService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class EmployeeServiceImpl implements EmployeeService {

    private final EmployeeRepository employeeRepository;

    @Override
    @Transactional(readOnly = true)
    public EmployeeDto getByCode(String code) {
        return employeeRepository.findByCode(code).map(EmployeeDto::toDto)
                .orElseThrow(() -> new EmployeeNotFoundException("Employee with " + code + " code not found!"));
    }

    @Override
    @Transactional(readOnly = true)
    public void auth(String code) {
        if (employeeRepository.findByCode(code).isEmpty()) {
            throw new EmployeeNotFoundException("Employee with " + code + " code not found!");
        }
    }
}

File: src/main/kotlin/ru/nto/storage/service/impl/EmployeeServiceImpl.kt
package ru.nto.storage.service.impl

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import ru.nto.storage.controller.dto.EmployeeDto
import ru.nto.storage.exception.EmployeeNotFoundException
import ru.nto.storage.repository.EmployeeRepository
import ru.nto.storage.service.EmployeeService

@Service
class EmployeeServiceImpl(
    private val employeeRepository: EmployeeRepository
) : EmployeeService {

    @Transactional(readOnly = true)
    override fun getByCode(code: String): EmployeeDto {
        val employee = employeeRepository.findByCode(code)
            ?: throw EmployeeNotFoundException("Employee with $code code not found!")
        return EmployeeDto.toDto(employee)
    }

    @Transactional(readOnly = true)
    override fun auth(code: String) {
        if (employeeRepository.findByCode(code) == null) {
            throw EmployeeNotFoundException("Employee with $code code not found!")
        }
    }
}
